app/src/test/kotlin/com/wire/android/ui/settings/devices/SelfDevicesViewModelTest.kt

```diff
@@ -44,19 +50,54 @@ import org.junit.jupiter.api.extension.ExtendWith
 class SelfDevicesViewModelTest {
 
     @Test
-    fun `given a self client id, when fetching self clients, then returns devices list without current device`() =
+    fun `given a self client id, when observing self clients, then returns devices list without current device`() =
         runTest {
             // given
+            val currentClient = TestClient.CLIENT
+            val otherClient = TestClient.CLIENT.copy(id = ClientId("anotherId"))
             val (_, viewModel) = Arrangement()
+                .withCurrentClientId(currentClient.id)
+                .withObserveClientsByUserIdResult(ObserveClientsByUserIdUseCase.Result.Success(listOf(currentClient, otherClient)))
                 .arrange()
 
             // when
-            val currentDevice = Device(TestClient.CLIENT)
+            val currentDevice = Device(currentClient)
+            val otherDevice = Device(otherClient)
 
             // then
-            assert(!viewModel.state.deviceList.contains(currentDevice))
+            assertEquals(false, viewModel.state.isLoadingClientsList)
+            assertEquals(SelfDevicesState.Error.None, viewModel.state.error)
+            assertEquals(false, viewModel.state.deviceList.contains(currentDevice))
+            assertEquals(true, viewModel.state.deviceList.contains(otherDevice))
         }
 
+    @Test
+    fun `given error, when observing devices, then show init error`() = runTest {
+        val (_, viewModel) = Arrangement()
+            .withFetchSelfClientsResult(SelfClientsResult.Failure.Generic(NetworkFailure.NoNetworkConnection(null)))
+            .withObserveClientsByUserIdResult(ObserveClientsByUserIdUseCase.Result.Failure(NetworkFailure.NoNetworkConnection(null)))
+            .arrange()
+        advanceUntilIdle()
+
+        assertEquals(false, viewModel.state.isLoadingClientsList)
+        assertEquals(SelfDevicesState.Error.InitError, viewModel.state.error)
+        assertEquals(emptyList(), viewModel.state.deviceList)
+    }
+
+    @Test
+    fun `given error, when retrying, then execute fetch properly`() = runTest {
+        val (arrangement, viewModel) = Arrangement()
+            .withFetchSelfClientsResult(SelfClientsResult.Failure.Generic(NetworkFailure.NoNetworkConnection(null)))
+            .arrange()
+        advanceUntilIdle()
+
+        viewModel.retryFetch()
+
+        coVerify(exactly = 2) { // first time during init, second time when retrying
```

Comment:
Could you please split it to make sure it is actually called the second time after the retry?

```kotlin
// ...
advanceUntilIdle()

coVerify(exactly = 1) { /** */ }

viewModel.retryFetch()

coVerify(exactly = 2) { /** */ } // Not sure if 1 or 2 here actually
```

Reply:
This test got removed because it's not needed after resolving comments from Mo 😅